packages/flame/lib/image_composition.dart

```diff
@@ -91,11 +91,13 @@ class ImageComposition {
     blendMode ??= defaultBlendMode;
     isAntiAlias ??= defaultAntiAlias;
 
-    assert(
-      imageRect.contains(source.topLeft) &&
-          imageRect.contains(source.bottomRight),
-      'Source rect should fit within in the image constraints',
-    );
+    // TODO(erick) unsure why, but this assertion seems to be incorrect
```

Comment:
@wolfenrain any idea on this?

Reply:
If you have an image that is 10 width and 10 height, and you pass a source where the bottomRight is (10, 10) then the contains will return false. This is something I faced as well, see: https://discord.com/channels/509714518008528896/538364802926379008/806580979103170601

Reply:
I have added a fix for that, please take a look